apps/server/src/modules/h5p-editor/contentStorage/contentStorage.spec.ts

```diff
@@ -1,471 +1,905 @@
+
+			return {
+				contentID,
+				deleteError,
+				filename,
+				invalidFilename,
+				user,
+			};
+		};
+
+		describe('WHEN deleting a file', () => {
+			it('should call S3ClientAdapter.delete', async () => {
+				const { contentID, filename, user } = setup();
+
+				await service.deleteFile(contentID, filename, user);
+
+				expect(s3ClientAdapter.delete).toHaveBeenCalledWith([expect.stringContaining(contentID)]);
 			});
 		});
 
-		describe('WHEN file can not be deleted', () => {
+		describe('WHEN filename is invalid', () => {
 			it('should throw error', async () => {
-				const { notExistingContentId, notExistingFilename } = setup();
-				try {
-					await service.deleteFile(notExistingContentId, notExistingFilename);
-				} catch (err) {
-					expect(err).toBeInstanceOf(Error);
-				}
+				const { contentID, invalidFilename, user } = setup();
+
+				const deletePromise = service.deleteFile(contentID, invalidFilename, user);
+
+				await expect(deletePromise).rejects.toThrow();
+			});
+		});
+
+		describe('WHEN S3ClientAdapter throws an error', () => {
+			it('should throw along the error', async () => {
+				const { contentID, filename, user, deleteError } = setup();
+				s3ClientAdapter.delete.mockRejectedValueOnce(deleteError);
+
+				const deletePromise = service.deleteFile(contentID, filename, user);
+
+				await expect(deletePromise).rejects.toBe(deleteError);
 			});
 		});
 	});
+
 	describe('fileExists', () => {
+		const setup = () => {
+			const filename = 'file.txt';
+			const invalidFilename = '..test.txt';
+
+			const deleteError = new Error('Could not delete');
+
+			const contentID = new ObjectID().toString();
+
+			return {
+				contentID,
+				deleteError,
+				filename,
+				invalidFilename,
+			};
+		};
+
 		describe('WHEN file exists', () => {
 			it('should return true', async () => {
-				const { contentId, filename1 } = setup();
-				const fileExists = await service.fileExists(contentId, filename1);
-				expect(fileExists).toEqual(true);
+				const { contentID, filename } = setup();
+				s3ClientAdapter.head.mockResolvedValueOnce(createMock());
+
+				const exists = await service.fileExists(contentID, filename);
+
+				expect(exists).toBe(true);
 			});
 		});
 
 		describe('WHEN file does not exist', () => {
 			it('should return false', async () => {
-				const { notExistingContentId, notExistingFilename } = setup();
-				const fileExists = await service.fileExists(notExistingContentId, notExistingFilename);
-				expect(fileExists).toEqual(false);
-			});
-		});
-		describe('WHEN conetntId is undefined', () => {
-			it('should throw an error', async () => {
-				const { undefinedContentId, notExistingFilename } = setup();
-				try {
-					await service.fileExists(undefinedContentId, notExistingFilename);
-				} catch (err) {
-					expect(err).toBeInstanceOf(Error);
-				}
+				const { contentID, filename } = setup();
+				s3ClientAdapter.head.mockRejectedValueOnce(new Error('NoSuchKey'));
```

Comment:
```suggestion
				s3ClientAdapter.head.mockRejectedValueOnce(new NotFoundException('NoSuchKey'));
```
Otherwise the test fails. You maybe overlooked because tests are only run in PRs into main.

Reply:
Changed for all instances